app/main.py

```diff
@@ -14,24 +14,34 @@ def load_settings():
 
 
 settings = load_settings()
-
-if __name__ == "__main__":
-    # Only for debugging while developing
-    app.run(host='0.0.0.0', debug=True, port=5000)
+yt = YouTrackHelper(settings["youtrack_instance_name"], settings["youtrack_token"])
 
 
 @app.route('/pull-request/', methods=['POST'])
 def assign():
+    users = settings["users_dict"]
     payload = request.get_json()
     pr = payload["pull_request"]
-    assignee = pr["assignee"]
-
-    if payload["action"] not in ["assigned", "opened"] or assignee is None:
-        return '', 200
-
     issue_id = pr["head"]["ref"]
-    assignee = settings["users_dict"][assignee["login"]]
     url = pr["url"]
-    yt = YouTrackHelper(settings["youtrack_instance_name"], settings["youtrack_token"])
-    yt.assign_ticket(issue_id, assignee, url)
+
+    if payload["action"] == "review_requested":
+        assignee = users[pr["requested_reviewers"][0]["login"]]
+        print(assignee)
+        yt.update_ticket(issue_id,
+                         commands=[yt.set_state("Submitted"), yt.assign(assignee)],
+                         comment=url)
+
+    if payload["action"] == "closed" and pr["merged"] is True:
```

Comment:
What about closed and *not* merged?

Reply:
I figured we don't get too opinionated about what to do in non-standard flows. Maybe it's a `Won't fix`, maybe there's going to be another PR for the same ticket so it's a `Reopened` and I'm sure there are other scenarios! So it's still on the person who closed the PR to update the ticket manually in this case.